src/Sylius/Behat/Page/Admin/Taxon/CreatePage.php

```
@@ -0,0 +1,134 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Page\Admin\Taxon;
+
+use Behat\Mink\Element\NodeElement;
+use Behat\Mink\Exception\ElementNotFoundException;
+use Sylius\Behat\Behaviour\SpecifiesItsCode;
+use Sylius\Behat\Page\Admin\Crud\CreatePage as BaseCreatePage;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class CreatePage extends BaseCreatePage implements CreatePageInterface
+{
+    use SpecifiesItsCode;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function countTaxons()
+    {
+        return count($this->getLeafs());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function countTaxonsByName($name)
+    {
+        $matchedLeafsCounter = 0;
+        $leafs = $this->getLeafs();
+        foreach ($leafs as $leaf) {
+            if ($leaf->getText() === $name) {
+                $matchedLeafsCounter++;
+            }
+        }
+
+        return $matchedLeafsCounter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function chooseParent(TaxonInterface $taxon)
+    {
+        $this->getElement('parent')->selectOption($taxon->getName(), false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteTaxonOnPageByName($name)
+    {
+        $leafs = $this->getLeafs();
+        foreach ($leafs as $leaf) {
+            if ($leaf->getText() === $name) {
+                $leaf->pressButton('Delete');
+
+                break;
+            }
+        }
```

I think we should throw exception when taxon with given name was not found.